src/btree.rs

```
@@ -283,10 +283,316 @@ impl BTree {
+            if current == old_key {
+                key_bytes.copy_from_slice(&new_key.to_le_bytes());
+                break;
+            }
+            offset += 16;
+        }
+
+        pager.write_page(parent_id, &parent_clone)?;
+        Ok(())
+    }
+
+    fn merge_leaves(&self, pager: &mut Pager, page_id: PageId, sibling_id: PageId, parent_id: PageId, separator_key: i64, root_page: PageId) -> Result<()> {
```

Clippy (1.95.0)

[warning] unused variable

(warning)